### PR TITLE
Adds LogEventAggregator with supportability metrics and tests

### LINKED ISSUE #907: Add LogEventAggregator
### Feature Description

----

**IMPORTANT**
Until PR #894 is merged, base branches off of `log-metrics` branch.
Merge any PRs into `logsisnt-feature-branch` branch

----

[Example code](https://github.com/newrelic/newrelic-dotnet-agent/blob/06a3a22acd1f3b4490f263b581e357d4a8ff2632/src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs)

- Add new aggregator “LogEventAggregator ” based on ErrorEventAggregator
- Uses ReaderWriterLockSlim and ConcurrentPriorityQueue with PrioritizedNode
- Add [LogEventAggregator ](https://github.com/newrelic/newrelic-dotnet-agent/blob/06a3a22acd1f3b4490f263b581e357d4a8ff2632/src/Agent/NewRelic/Agent/Core/DependencyInjection/AgentServices.cs#L127)to DI in AgentServices
- Add supportability ([AgentHealthReporter](https://github.com/newrelic/newrelic-dotnet-agent/blob/06a3a22acd1f3b4490f263b581e357d4a8ff2632/src/Agent/NewRelic/Agent/Core/AgentHealth/AgentHealthReporter.cs#L668-L673)): ReportLoggingEventCollected, ReportLoggingEventsRecollected, ReportLoggingEventsSent  (  [more links ](https://github.com/newrelic/newrelic-dotnet-agent/blob/06a3a22acd1f3b4490f263b581e357d4a8ff2632/src/Agent/NewRelic/Agent/Core/WireModels/MetricWireModel.cs#L958-L1005))
- Supportability [includes setting up the metrics properly in MetricNames](https://github.com/newrelic/newrelic-dotnet-agent/blob/06a3a22acd1f3b4490f263b581e357d4a8ff2632/src/Agent/NewRelic/Agent/Core/NewRelic.Agent.Core.Metric/MetricNames.cs#L487-L491) and other places.
- Add unit test is possible/logical

Acceptance Criteria

- New unit tests for supportability pass

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

- Adds LogEventsAggregator using ErrorEventAggregator as a reference
- Adds new supportability metrics in AgentHealthReporter to track was is collected, sent, and recollect
- Adds tests for new supportability metrics
- Adds new test class for LogEventAggregator using ErrorEventAggregatorTests as a reference

Resolves #907

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)
- [ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [x] Perform code review
- [x] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+            _readerWriterLock.EnterWriteLock();
+            try
+            {
+                originalLogEvents = GetAndResetLogEvents(GetReservoirSize());
+            }
+            finally
+            {
+                _readerWriterLock.ExitWriteLock();
+            }
```
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs:100]: My experience with InterlockedExchange is that you do not need this extra layer of locking. Remove the ReaderWriterLock, and this should work as expected.

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
I think we can eliminate the extra lock here

File: src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs (hunk 1 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Core/Aggregators/LogEventAggregator.cs]: Pushed up a commit with locking removed.

**Review by JcolemanNR — APPROVED** [review]
I like all the tests!

**Pull request merged**